.travis.yml

```diff
@@ -0,0 +1,45 @@
+dist: trusty
+
+cache:
+  apt: true
+  directories:
+    # Cache directory for older Composer versions.
+    - $HOME/.composer/cache/files
+    # Cache directory for more recent Composer versions.
+    - $HOME/.cache/composer/files
+
+language:
+  - php
+
+php:
+  - 5.6
+  - 7.0
+  - 7.1
+  - 7.2
+  - 7.3
+  - "7.4snapshot"
+
+matrix:
+  fast_finish: true
+  include:
```

Comment:
We don't need to run linting on every version of PHP, just run it on PHP 7.3, and add the `script` to the `matrix`.

Also, if this PR is just focussed on adding linting and Travis (but not fixing the linting problems), it should mark the linting job in `allowed_failures`.

Reply:
So basically, just keep the 7.3 version. Not sure I understand the 'add `script` to the `matrix`' part.

> Also, if this PR is just focussed on adding linting and Travis (but not fixing the linting problems), it should mark the linting job in allowed_failures.

Well, this is the tricky bit. I'd want to enforce the code to pass the PHPCS checks, at the same time, the current PR will clearly fail because I don't want to make changes and add the configuration in the same PR (I'd have tons of changed files that should be checked). If I set it as `allowed_failures` the PR could be merged, but it should be reverted on the other PR that will then fix the issues (so that code checks are included).

I thought that once the configuration is ok, it can be merged (despite it failing on Travis) and then a new PR will be made with just changes from the PHPCS fixes.

Reply:
As for the workflow: I'd suggest opening the PR(s) fixing CS failures concurrently. Once those have been merged, this PR can be rebased and the build should then pass.

Pro-tip: Small PRs targetting specific CS failures are easier to review than having one large PR fixing everything in one go.
This also allows for discussions on which rules should or should not be excluded to be had separately instead of all in one go.